uncontrast_studies/management/commands/process_uncon_row.py

```diff
@@ -0,0 +1,121 @@
+
+from contrast_api.choices import ExperimentTypeChoices
+from contrast_api.data_migration_functionality.studies_parsing_helpers import ProblemInStudyExistingDataException
+from studies.models import Study
+from uncontrast_studies.models import UnConExperiment, UnConSpecificParadigm, UnConTask, UnConSuppressedStimulus, \
+    UnConTargetStimulus, UnConSuppressionMethodType, UnConSuppressionMethodSubType, UnConSuppressionMethod, UnConSample, \
+    UnConProcessingMainDomain, UnConProcessingDomain, UnConFinding, UnConsciousnessMeasurePhase, \
+    UnConsciousnessMeasureType, UnConsciousnessMeasureSubType, UnConsciousnessMeasure, UnConTaskType
+from uncontrast_studies.parsers.consciousness_measure_parser import resolve_consciousness_measures
+from uncontrast_studies.parsers.uncon_data_parsers import resolve_uncon_paradigm, resolve_uncon_task, resolve_uncon_processing_domains
+from uncontrast_studies.parsers.stimulus_parser import resolve_uncon_stimuli, resolve_uncon_stimuli_metadata
+from uncontrast_studies.parsers.suppression_method_parser import resolve_uncon_suppression_method
+from uncontrast_studies.parsers.sample_parser import resolve_uncon_samples
+from uncontrast_studies.parsers.finding_parser import resolve_uncon_findings
+
+ExperimentDuplicate = namedtuple("ExperimentDuplicateTuple", ["key", "id"])
+
+
+def create_uncon_experiment(item: dict, index):
+    try:
+        if item["Paper.DOI"]:
+            study = Study.objects.get(DOI=item["Paper.DOI"])
+        else:
+            study = Study.objects.get(DOI=item["StudyID"])  # TODO: change back to DOI
+    except ObjectDoesNotExist:
+        raise ProblemInStudyExistingDataException()
+
+    paradigm_data = resolve_uncon_paradigm(item, index)
+    paradigm, created = UnConSpecificParadigm.objects.get_or_create(
+        main=paradigm_data.main, name=paradigm_data.specific
+    )
+    stimuli_metadata = resolve_uncon_stimuli_metadata(item, index)
+
+    experiment, created = UnConExperiment.objects.get_or_create(
+        study=study,
+        type=ExperimentTypeChoices.BEHAVIORAL,
+        paradigm=paradigm,
+        is_target_stimulus=stimuli_metadata.is_target_stimulus,
+        is_target_same_as_suppressed_stimulus=stimuli_metadata.is_target_same_as_prime,
+    )
+
+    return experiment
+
+
+def process_uncon_row(item: dict):
+    experiment_index = item["exp"]
+    experiment = create_uncon_experiment(item=item, index=experiment_index)
+
+    # tasks
+    task_types = resolve_uncon_task(item=item, index=experiment_index)
+    for task_type in task_types:
+        task, created = UnConTaskType.objects.get_or_create(name=task_type)
+        UnConTask.objects.get_or_create(experiment=experiment, type=task)
+
+    # samples
+    samples = resolve_uncon_samples(item=item, index=experiment_index)
+    for sample in samples:
+        UnConSample.objects.get_or_create(experiment=experiment, type=sample.sample_type, size_included=sample.included_size,
```

Comment:
I don't think it's a get or create - afaik there might be two samples for the same experiment with the exact same detail. no?